lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -19,12 +19,14 @@ class CheckoutV2Gateway < Gateway
       def initialize(options = {})
         @options = options
         @access_token = nil
-        begin
+
+        if options.has_key?(:secret_key)
           requires!(options, :secret_key)
-        rescue ArgumentError
+        else
           requires!(options, :client_id, :client_secret)
           @access_token = setup_access_token
         end
+
```

Comment:
I think this might be the problematic change.

What happens with `requires!` is that it raises an `ArgumentError` which was being `rescue`d previously in the case where a user was using the OAuth authentication.

Reply:
Sorry, this is not the root cause of the error we encountered